File: sem4/MPP/proiect/appClientServer/CompetitionServices/src/main/java/org/example/service/ICompetitionServices.java
package org.example.service;

import org.example.model.*;

public interface ICompetitionServices {
    User login(User user, ICompetitionObserver client) throws CompetitionException;
    TaskDTO[] findAllTasks() throws CompetitionException;
    Child[] findByTask(Task task) throws CompetitionException;
    void addParticipation(Participation participation) throws CompetitionException;
    void addChild(Child kid) throws CompetitionException;
    Child findLastAdded() throws CompetitionException;
    void logout(User user, ICompetitionObserver client) throws CompetitionException;
    void updateEvent() throws CompetitionException;
}

File: sem4/MPP/proiectC#/AppClientServer/Services/ICompetitionServices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using model;

namespace services
{
    public interface ICompetitionServices
    
    {
        void login(User user, ICompetitionObserver client);
        List<TaskDTO> findAllTasks();
        List<Child> findByTask(Task task);
        void addParticipation(Participation participation);
        void addChild(Child child);
        Child findLastAdded();
        
        void logout(User user, ICompetitionObserver client);
        void updateEvent();

    }
}
